### PR TITLE
#644 add samples for arrayAssertions of api-fluent

### LINKED ISSUE #644: add samples for arrayAssertions of api-fluent
## Code related feature
See for the initial request: https://github.com/robstoll/atrium/issues/624

With this issue we want to add samples for arrayAssertions.kt of api-fluent

Following the things you need to do:
 
*api-fluent*
- [ ] create ArrayAssertionSamples in apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples
  - [ ] add one function per function in arrayAssertions.kt (see AnyAssertionSamples)
  - [ ] refer to the sample in the KDoc of the corresponding function in arrayAssertions via `@sample` (check out `toBe` in anyAssertions.kt to see how you have to refer to the sample)

## Your first contribution?
- Write a comment `I'll work on this` if you would like to take this issue over. 
  This way we get the chance to revise the description in case things have changed in the meantime, we might give you additional hints and we can assign the task to you, so that others do not start as well.
- See [Your first code contribution](https://github.com/robstoll/atrium/blob/master/.github/CONTRIBUTING.md#your-first-code-contribution) for guidelines.  
- Do not hesitate to ask questions here or to contact us via [Atrium's slack channel](https://kotlinlang.slack.com/team/U3DE1TXKP) if you need help
  ([Invite yourself](https://slack.kotlinlang.org/) in case you do not have an account yet).

### PR DESCRIPTION
**Pull request by jdornieden**
Resolve #644 
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+import ch.tutteli.atrium.api.fluent.en_GB.toBe
+import ch.tutteli.atrium.api.verbs.internal.expect
+import kotlin.test.Test
+
+class ArrayAssertionSamples {
+
+    @Test
+    fun asListFeature() {
+        expect(arrayOf("A", "B"))
+            .asList()
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt:14]:
```suggestion
            .asList()  // subject is now of type List<String>
```

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            .toBe(listOf("A", "B"))
+    }
+
+    @Test
+    fun asList() {
+        expect(arrayOf("A", "B"))
+            .asList {
+                toBe(listOf("A", "B"))
+            }
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt:23]:
```suggestion
            .asList { // subject within this block is of type List<String>
                toBe(listOf("A", "B"))
            } // subject here is back to type Array<String>
```

**Review by robstoll — CHANGES_REQUESTED** [review]
Could you please also add the two comments to all other functions and adapt? Thanks

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt (hunk 2 above)
Comment (jdornieden) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt:23]: Sure!

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                }
+        }
+    }
+
+    @Test
+    fun asListEOut() {
+        expect<Array<out String>>(arrayOf("A", "B"))
+            .asList { // subject within this block is of type List<String>
+                toBe(listOf("A", "B"))
+            } // subject here is back to type Array<String>
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt:38]:
```suggestion
            } // subject here is back to type Array<out String>
```
No?

Comment (jdornieden) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/ArrayAssertionSamples.kt:38]: Yes! :-)

**Review by robstoll — APPROVED** [review]

**Comment by robstoll** [comment]
@jdornieden thanks for your 2nd contribution 👍
Looking forward to your third

**Pull request merged**